instrumentation/spring-webmvc-3.1/src/main/java/io/opentelemetry/auto/instrumentation/springwebmvc/DispatcherServletInstrumentation.java

```diff
@@ -131,10 +134,10 @@ public static void stopSpan(
     @Advice.OnMethodEnter(suppress = Throwable.class)
     public static void nameResource(@Advice.Argument(3) final Exception exception) {
       final Span span = TRACER.getCurrentSpan();
-      if (span.getContext().isValid() && exception != null) {
+      if (span.getContext().isValid()) {
         // We want to capture the stacktrace, but that doesn't mean it should be an error.
         // We rely on a decorator to set the error state based on response code. (5xx -> error)
-        DECORATE.addThrowable(span, exception);
+        DECORATE.onError(span, exception);
```

Comment:
`onError()` contains a null check. If `exception == null` then it will run addThrowable().